test/behaviour/graql/GraqlSteps.java

```diff
@@ -104,7 +109,7 @@ public void graql_insert_throws(String insertQueryStatements) {
 
     @Given("graql insert; throws exception containing {string}")
     public void graql_insert_throws_exception(String exception, String insertQueryStatements) {
-        assertThrowsWithMessage(() -> graql_insert(insertQueryStatements), exception);
+        assertThrowsWithMessage(() -> graql_insert(insertQueryStatements).iterator().next(), exception);
```

Comment:
The previous implementation had a bug that caused it to always fail. `QueryManager.insert` is NOT supposed to throw an error; _iterating_ over it is.

Reply:
if it doesn't throw on insert, itll throw on commit right? (just double checking)

Reply:
It _does_ throw on insert. But this is Client Java - `insert` uses deferred execution; it sends the request up to the server and immediately returns a `Stream`. The user sees the exception only if they iterate over the Stream.